### DISCUSSION TITLE
Error: must run as root

### DISCUSSION THREAD

**Discussion by rabol** [deployphp/deployer#2847] [Help needed]
Hi

I'm trying to provision a new server using deployer

after I have provided information the provisioning stops more or less immediately

Error: must run as root
error  in provision.php on line 83:
exit code 1 (General error)

deployer version is : Deployer 7.0.0-rc.3

I'm trying to provision a server on a vps from webdock.io
I have created a admin user that I use, added public key and even set 'passwordless sudo for all users in sudo group'

**Comment by antonmedv** [comment]
Provision recipe designed to be executed as root. No sudo commands was added. 

For example next command can be executed only as root: apt-add-repository

**Reply by rabol** [reply]
When I provision the server with only the OS I don't get the root, i can create any user I want, but not root

**Reply by antonmedv** [reply]
Ubuntu has a root.

**Reply by rabol** [reply]
yes, I managed to enable root, but it would be cool if you could have another user
e.g with passwordless sudo enabled

**Reply by rabol** [reply]
hmm

The command "export DEBIAN_FRONTEND='noninteractive'; apt-add-repository ppa:ondrej/php -y" exceeded the timeout of 300 seconds.

if I do that manually 
Error: retrieving gpg key timed out.

**Reply by antonmedv** [reply]
Looks like ondrej/php having some troubles.
